src/WebJobs.Script.WebHost/Controllers/FunctionsController.cs

```diff
@@ -226,7 +226,7 @@ private async Task ScheduleRestartAsync(IFileMonitoringService fileMonitoringSer
             Task restartTask = null;
             using (System.Threading.ExecutionContext.SuppressFlow())
             {
-                restartTask = fileMonitoringService.ScheduleRestartAsync(false);
+                restartTask = fileMonitoringService.RestartIfScheduledAsync();
```

Comment:
The containing method name no longer matches the logic here, which is a bit confusing. However, maybe this method can just go away, as per suggestion below.